pkg/api/types.go

```diff
@@ -173,9 +173,9 @@ type VolumeSource struct {
 	HostPath *HostPathVolumeSource `json:"hostPath"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDirVolumeSource `json:"emptyDir"`
-	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// CloudPersistentDisk represents a cloud-provided storage resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
-	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+	CloudPersistentDisk *CloudPersistentDiskVolumeSource `json:"persistentDisk"`
```

Comment:
@justinsb could you modify this so it implements AWS persistent volumes as sibling to GCE PD. Similar to other proposed persistent volume plugins: #4612, and #4601.
I think there is general consensus on doing that for now, then improving the abstraction to promote code sharing after a couple of them are merged. See the discussion here on  https://github.com/GoogleCloudPlatform/kubernetes/pull/4601/files#r25993686